pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -61,37 +64,37 @@ func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 		return true, nil, nil
 	}
 
-	nodeVolumeLimits := nodeInfo.VolumeLimits()
-
-	// if node does not have volume limits this predicate should exit
-	if len(nodeVolumeLimits) == 0 {
-		return true, nil, nil
-	}
-
 	// a map of unique volume name/csi volume handle and volume limit key
 	newVolumes := make(map[string]string)
-	if err := c.filterAttachableVolumes(pod.Spec.Volumes, pod.Namespace, newVolumes); err != nil {
+	if err := c.filterAttachableVolumes(nodeInfo, pod.Spec.Volumes, pod.Namespace, newVolumes); err != nil {
 		return false, nil, err
 	}
 
 	if len(newVolumes) == 0 {
 		return true, nil, nil
 	}
 
+	nodeVolumeLimits := nodeInfo.VolumeLimits()
```

Comment:
Why can't we have this check earlier and exit earlier?

Reply:
I'm using this map to verify if a potential CSI driver is installed on the node and report the appropriate error.